conflowgen/logging/logging.py

```diff
@@ -52,11 +52,15 @@ def setup_logger(
     if logging_directory is None:
         logging_directory = LOGGING_DEFAULT_DIR
 
+    logger = logging.getLogger("conflowgen")  # Check if logger already exists
+    if logger.handlers:
```

Comment:
Is that part of the official API? I cannot find it on https://docs.python.org/3/library/logging.html. However, the method `Logger.hasHandlers` looks official. Then you could also specify which handler you want to look for.

Why this is important: Somebody might use this library but wants a different style of logging. These people could first invoke some code like `logger = logging.getLogger("conflowgen")` themselves, then add their custom handlers to that (maybe streaming some content over a socket to another server) and only afterwards invoke our helper function to do the remaining setup. Your proposed approach might not harmonize well with that because you prematurely interrupt the setup.

I would rather prefer that we only avoid having two StreamHandlers streaming to stdout. This requires some more checks then there are currently in place. The field `stream` of the StreamHandler is not part of the documented API but as there is no leading underscore, it should not change over the upcoming versions either (see https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L1139 for reference). So you could check whether two StreamHandlers stream to stdout and only stop that? In addition, you might write a warning to the logs so that people are not confused when things do not work out as planned. I assume warnings are better because an Exception would be too harsh.